### PR TITLE
refactor(anta): Refactor VerifyDNSServers test for nicer failure message 

### PR DESCRIPTION
**Pull request by vitthalmagadum**
# Description

Refactor VerifyDNSServers test for nicer failure message 

Related to #587 #388 

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have run pre-commit for code linting and typing (`pre-commit run`)
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: anta/tests/services.py [hunk 1]
```diff
@@ -136,37 +143,31 @@ class Input(AntaTest.Input):
         dns_servers: list[DnsServer]
         """List of DNS servers to verify."""
 
-        class DnsServer(BaseModel):
```
Comment (carl-baillargeon) [inline anta/tests/services.py:139]: Need to add a `ClassVar` to keep existing behavior.

File: anta/tests/services.py [hunk 2]
```diff
@@ -106,10 +106,17 @@ def test(self) -> None:
 class VerifyDNSServers(AntaTest):
     """Verifies if the DNS (Domain Name Service) servers are correctly configured.
 
+    This test performs the following checks for each specified LLDP neighbor:
```
Comment (carl-baillargeon) [inline anta/tests/services.py:109]:
```suggestion
    This test performs the following checks for each specified DNS server:
```

File: anta/tests/services.py [hunk 3]
```diff
@@ -136,37 +143,31 @@ class Input(AntaTest.Input):
...
             if get_item(command_output, "ipAddr", address) is None:
-                self.result.is_failure(f"DNS server `{address}` is not configured with any VRF.")
+                self.result.is_failure(f"Server {address} - Not configured with any VRF")
                 continue
 
+            # Check if the DNS server is configured with specified VRF.
             if (output := get_dict_superset(command_output, input_dict)) is None:
-                self.result.is_failure(f"DNS server `{address}` is not configured with VRF `{vrf}`.")
+                self.result.is_failure(f"{server} - Not configured in specified VRF")
                 continue
```
Comment (carl-baillargeon) [inline anta/tests/services.py:166]: I think we can simplify this by having a single check for the IP + VRF.

File: anta/tests/services.py (hunk 2 above)
Comment (vitthalmagadum) [reply anta/tests/services.py]: Updated

**Review by carl-baillargeon — APPROVED** [review]
LGTM

**Pull request merged**